CONTRIBUTING.md

```diff
@@ -20,14 +20,84 @@
 - Avoid adding third-party dependencies, extra files, extra headers, etc.
 - Always consider cross-compatibility with other operating systems and architectures
 - Avoid fancy-looking modern STL constructs, use basic `for` loops, avoid templates, keep it simple
-- There are no strict rules for the code style, but try to follow the patterns in the code (indentation, spaces, etc.). Vertical alignment makes things more readable and easier to batch edit
+- Vertical alignment makes things more readable and easier to batch edit
 - Clean-up any trailing whitespaces, use 4 spaces for indentation, brackets on the same line, `void * ptr`, `int & a`
-- Naming usually optimizes for common prefix (see https://github.com/ggerganov/ggml/pull/302#discussion_r1243240963)
+- Use sized integer types in the public API
+- Declare structs with `struct foo {}` instead of `typedef struct foo {} foo`
+    - In C++ code omit the `struct` keyword whenever it is not necessary
```

Comment:
In my code I intentionally did it the other way around but I would be fine with either way. Does the same principle apply to enums?

Reply:
Yes, I've been also using it a lot and actually prefer to have the `struct` and `enum` keywords explicitly stated everywhere like in C. Unfortunately, there is no way to enforce this in C++ and removing a keyword is much easier than finding where it wasn't added, so to keep it simple, I think we should remove all optional keywords.

Reply:
It's just very uncommon in C++ code to use `struct` `enum` or `class` keywords in variable declarations, unless you explicitly want to add a new type that isn't declared before.